src/VisualStudio/Core/Def/Implementation/ProjectSystem/VisualStudioWorkspaceImpl.OpenFileTracker.cs

```diff
@@ -83,13 +91,35 @@ public sealed class OpenFileTracker : IRunningDocumentTableEventListener
             /// This cutoff of 10 was chosen arbitrarily and with no evidence whatsoever.</remarks>
             private const int CutoffForCheckingAllRunningDocumentTableDocuments = 10;
 
-            private OpenFileTracker(VisualStudioWorkspaceImpl workspace, IVsRunningDocumentTable runningDocumentTable, IComponentModel componentModel)
+            private OpenFileTracker(
+                VisualStudioWorkspaceImpl workspace,
+                IThreadingContext threadingContext,
+                IVsRunningDocumentTable runningDocumentTable,
+                IComponentModel componentModel,
+                IAsynchronousOperationListenerProvider listenerProvider)
             {
                 _workspace = workspace;
+                _threadingContext = threadingContext;
                 _foregroundAffinitization = new ForegroundThreadAffinitizedObject(workspace._threadingContext, assertIsForeground: true);
                 _asyncOperationListener = componentModel.GetService<IAsynchronousOperationListenerProvider>().GetListener(FeatureAttribute.Workspace);
```

Comment:
Remove the IComponentModel parameter now that your'e passing in the IAsynchronousOperationListenerProvider directly?

Reply:
i totally missed we had an asyncOpLIstener already.  i reverted a byunch so i could just use that.